### PR TITLE
TREE: Match must have an argument 

### PR DESCRIPTION
**Pull request by miripiruni**
#418 fixed

### CODE REVIEW

**Comment by miripiruni** [comment]
cc @yalovek

File: lib/bemxjst/tree.js [hunk 1]
```diff
@@ -353,6 +353,10 @@ Tree.prototype.body = function() {
 
 Tree.prototype.match = function() {
   var children = new Array(arguments.length);
+
+  if (!arguments.length)
+    throw new Error('.match() must have argument');
```
Comment (zxqfox) [inline lib/bemxjst/tree.js]: `if (!something) throw new Error('error')` is a pure assert pattern: `assert(something, 'error')`
prob we can get some bytes on it.

Comment (miripiruni) [inline lib/bemxjst/tree.js]: after gzip:

term 1: `throw new Error('`
term 2: `')`

In your case:

term 1: `assert(something, '`
term 2: `')`

Same shit…

Comment (zxqfox) [inline lib/bemxjst/tree.js]: For the point point of view: you dont need test case for `assert(arguments.length)`

Comment (miripiruni) [inline lib/bemxjst/tree.js]: @zxqfox m?

Comment (zxqfox) [inline lib/bemxjst/tree.js]: nvm, anyway you writing tests for exceptions

Comment (tadatuta) [inline lib/bemxjst/tree.js]: it should be **an** argument but I'm not sure if it's significant in this case

**Pull request merged**